### ISSUE TITLE
Don't take exit codes into account

### ISSUE THREAD

**Issue by ScoreUnder** [jarv/cmdchallenge#73]
---

---

# Warning: Challenge spoilers

---

---

Often an exit code does no harm, and the task successfully completes despite the tool used reporting an error.
For example:
```bash
bash(0)> find \! \( -name \*.exe -o -name \*.txt \) -delete
+ find '!' '(' -name '*.exe' -o -name '*.txt' ')' -delete
find: cannot delete './b': Directory not empty
find: cannot delete './c/m': Directory not empty
find: cannot delete './c': Directory not empty
bash(1)> find \! \( -name \*.exe -o -name \*.txt \) -delete||:
# 👍 👍 👍  Correct!
```
(Note: The cmdchallenge software seems to have squashed stderr when the command succeeded)
On a tangential note, if this specific case is meant to be a failure, perhaps you could put an empty directory somewhere for that command to accidentally delete.

Another time I ran into this was when writing a loop for the `search_for_files_containing_string` problem.
I solved it with something like:
```bash
for f in ./*; do grep -q 500 "$f" && printf %s\\n "${f##*/}"; done
```
This failed until I added the `||:` at the end to squash the exit code.

The exit code of a for-loop is usually not very meaningful (unless you go out of your way to make it so), and as with the previous command, the presence of a nonzero exit code doesn't affect the correct operation of the script.

**Comment by jarv** [comment]
Thanks for the bug report, this is already fixed on master with 3539dd621aba9fab99f93dadd18c687a935a7409 and will go live today, I will update this issue when it is on the website.

**Comment by jarv** [comment]
This is now live on the site, please let me know if you continue to see problems and thank you very much for the bug report!
